contracts/examples/SimpleInterestTermsContract.sol

```diff
@@ -135,21 +135,25 @@ contract SimpleInterestTermsContract is TermsContract {
             uint120 _termLengthInAmortizationUnits
         )
     {
-        bytes16 principalPlusInterest;
-        bytes1 amortizationUnitType;
-        bytes15 termLengthInAmortizationUnits;
-
-        // In Solidity, the only way by which we can granularly split a 32 byte
-        // field into its constituent components is via inline assembly.
-        // We split the first 16 bytes of the parameters into the hex
-        // encoded principal plus interest, the subsequent byte into
-        // the hex encoded amortization unit type code, and the remining
-        // 15 bytes into the hex encoded term length.
-        assembly {
-            principalPlusInterest := calldataload(4)
-            amortizationUnitType := calldataload(20)
-            termLengthInAmortizationUnits := calldataload(21)
-        }
+        // The first 16 bytes of the parameters represent the total principal + interest
```

Comment:
Really happy to see us move out of assembly. Although this still feels pretty low level. Hoping that Ethereum will add better abstractions for this sort of functionality.